js/app.js

```diff
@@ -135,42 +177,93 @@ function handleUpdateRateForm(event) {
     .value.trim()
     .toUpperCase();
 
-  // Get the value of the new exchange rate from the input field and convert it to a float
+  // Get the value of the new exchange rate from the input field
   const newExchangeRate = parseFloat(
     document.getElementById("update-exchange-rate").value
   );
 
-  // Find the rate object for the base currency
-  const rateObject = currencyRates.find((rate) => rate.base === baseCurrency);
+  // Find the rate
+  const rate = currencyRates.find(
+    (rate) =>
+      rate.baseCurrency === baseCurrency &&
+      rate.targetCurrency === targetCurrency
+  );
 
-  if (rateObject && rateObject.rates[targetCurrency]) {
-    // If the rate object and the exchange rate for the target currency are found
-    rateObject.rates[targetCurrency] = newExchangeRate; // Update the exchange rate
+  // If rate not found, show an error message
+  if (!rate) {
+    displayMessage(
+      `Exchange rate from ${baseCurrency} to ${targetCurrency} not found.`,
+      false
+    );
+    return;
+  }
 
-    // Log the updated exchange rates array to the console
-    console.log("Updated exchange rates:", currencyRates);
+  // Update the rate
+  rate.exchangeRate = newExchangeRate;
 
-    // Display success message
-    displayMessage("Exchange rate successfully updated!", true);
-  } else {
-    // If no rate is found for the update
-    console.log("Rate not found. Please add it first.");
+  // Reset the form fields
+  event.target.reset();
+
+  // Update the displayed rates
+  renderCurrencyGrid();
+
+  // Show success message
+  displayMessage(
+    `Exchange rate from ${baseCurrency} to ${targetCurrency} updated successfully!`
+  );
+  console.log(
+    `Exchange rate from ${baseCurrency} to ${targetCurrency} updated successfully!`
+  );
+}
+
+// Function to handle search form
+function handleSearchForm(event) {
+  event.preventDefault();
+
+  const fromCurrency = document
+    .getElementById("search-from")
+    .value.trim()
+    .toUpperCase();
+  const toCurrency = document
+    .getElementById("search-to")
+    .value.trim()
+    .toUpperCase();
+
+  const rate = currencyRates.find(
+    (rate) =>
+      rate.baseCurrency === fromCurrency && rate.targetCurrency === toCurrency
```

Comment:
This should be OR instead of AND. Be careful to read the brief with attention. It says `searching by the currency from OR to`

```suggestion
      rate.baseCurrency === fromCurrency || rate.targetCurrency === toCurrency
```

Reply:
Hi Valentin, thank you. It works strangely and I can't figure out how to fix it.
<img width="600" alt="image" src="https://github.com/user-attachments/assets/1d9fbf28-702e-491f-8409-3c21423cbcd8">
<img width="599" alt="image" src="https://github.com/user-attachments/assets/73f4441e-7bcc-44a8-8e37-c2659cc61215">

Reply:
Check out my message on Slack in #js-currency-converter-team29. You are not the only one having issues with the search, so I have written some details about the expected behaviour.

Reply:
Could you review please? I guess it's working